query-planner-js/src/snapshotSerializers/queryPlanSerializer.ts

```diff
@@ -62,21 +73,48 @@ function printNode(
             ' =>' +
             config.spacingOuter
           : '') +
-        printer(
-          flattenEntitiesField(parse(node.operation)),
-          config,
-          indentationNext,
-          depth,
-          refs,
-          printer,
-        ) +
+        printOperation(node.operation, indentationNext) +
         config.spacingOuter +
         indentation +
         '}';
       break;
     case 'Flatten':
       result += `Flatten(path: "${node.path.join('.')}")`;
       break;
+    case 'Defer':
+      const primary = node.primary;
+      const indentationInner = indentationNext + config.indent;
+      result +=
+        'Defer {' + config.spacingOuter +
+        indentationNext + `Primary {` + config.spacingOuter +
+        (primary.subselection ? printOperation(primary.subselection, indentationInner): indentationInner) + ':' + config.spacingOuter +
+        (primary.node ? (indentationInner + printNode(primary.node, config, indentationInner, depth, refs, printer) + config.spacingOuter) : '') +
+        indentationNext + '}, [' +
+        printDeferredNodes(node.deferred, config, indentationNext, depth, refs, printer) +
+        ']' + config.spacingOuter +
+        indentation + '}';
+      break;
+    case 'Condition':
+      if (node.ifClause) {
+        if (node.elseClause) {
+          result +=
+            `Condition(if: \$${node.condition}) {` + config.spacingOuter +
+            indentationNext + printNode(node.ifClause, config, indentationNext, depth, refs, printer) + ',' + config.spacingOuter +
+            indentationNext + printNode(node.elseClause, config, indentationNext, depth, refs, printer) + config.spacingOuter +
```

Comment:
I'd find the tests with if/else easier to read if the serializer explicitly wrote "Else" or something in it

Reply:
fixed in #2093, thanks